### ISSUE TITLE
[FR] A custom scriptable GLSL effect

### ISSUE THREAD

**Issue by unfa** [olive-editor/olive#692]
I thought I'd be nice to have a "Custom" video effect that'd expose an editable shader text that the user could experiment with seeing immediate visual effects. It could be great for also exchanging effects on forums or prototyping effects for Olive.

Providing a basic "no-op" template would be also nice.

What do you think?

**Comment by alcomposer** [comment]
This usability is partly already in Olive. You would need to make an XML file pointing to your shader code. There are default examples of this is th effects folder IIRC.

Only thing I that doesn't exist is an editor. But considering how many good GLSL editors are out there (and even in the browser) is this really needed?

**Comment by alcomposer** [comment]
Here is an example, un-zip it and place it in the `effects/shaders` directory.

[wheel_test.zip](https://github.com/olive-editor/olive/files/3007410/wheel_test.zip)

**Comment by unfa** [comment]
Thanks a lot!

**Comment by alcomposer** [comment]
However there would be one reason to have a basic text editor in Olive, due to the internal use of scene referred values, the output of shaders could be different. However a user could edit shaders while Olive is running, and update the shader in Olive when the code has changed.

**Comment by cgvirus** [comment]
AWESOME! Finally an editor that can save the world! This is going to be one of the most advance editor in near future. Thanks @alcomposer  and @unfa  for the heads up!

**Comment by cgvirus** [comment]
Just made a pull request for some glitch effect. Let's see how things go! I love this app.

**Comment by frink** [comment]
Is this a dupe of: #586 ? At the very least it's closely related.

**Comment by cgvirus** [comment]
@itsmattkc  Hi, I was investigating the current pipeline of node creation in 0.2 yesterday and it seems we now need to compile nodes within C. Will the previous pipeline going to be integrated? Thanks.

**Comment by itsmattkc** [comment]
Not exactly. The point of nodes is that (when there are more of them implemented), effects can be wholly constructed in-app. For easy sharing, node chains created in Olive can simply be copied and pasted to/from text (which you can experiment with now). Finally, there will be a GLSL shader node as described in this issue if you need to write shader code, which will obviously be included when copying/pasting nodes to/from text.

**Comment by cgvirus** [comment]
This is super nice! Just tested copy pasting. Excited to see the GLSL node soon! Thanks!
